### PR TITLE
Feature/color hist

### PR DESCRIPTION
**Pull request by alxrsngrtn**
This PR consists of the Color Labeling Command. I used histogram binning to break down colors into 10 categories, which I store in the state objects in the `state_q` buffer. 

Included in this change: 
- Working end to end color detector that labels the primary colors of one of the objects in view (presented in demo today). 
- Updated how we persist state for the application overall. Added `update_state_dict` function that deprecates the `update_state` function. 

EDIT: 
- Also updated MQTT topics. Now we can communicate with the agent and explorer. 

Reviewers: 
@hobson @ashwinkannan94

### CODE REVIEW

File: nlp/command/color.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from nlp import describe_state
+from nlp.dispatch import Dispatchable
+
+
+class DescribeColor(Dispatchable):
+
+    color_tmpl = "The {} is primarily {}"
+
```
Comment (hobson) [inline nlp/command/color.py]: named template variables

Comment (alxrsngrtn) [reply nlp/command/color.py]: Made the change on my local. I'll push it up in a jiffy.
